src/Verify.MSTest.SourceGenerator/Emitter.cs

```diff
@@ -0,0 +1,79 @@
+static class Emitter
+{
+    static readonly string AutoGenerationHeader = """
+        //-----------------------------------------------------
+        // This code was generated by a tool.
+        //
+        // Changes to this file may cause incorrect behavior
+        // and will be lost when the code is regenerated.
+        // <auto-generated />
+        //-----------------------------------------------------
+        """;
+
+    static readonly string GeneratedCodeAttribute =
+        $"[global::System.CodeDom.Compiler.GeneratedCodeAttribute(\"{typeof(Emitter).Assembly.GetName().Name}\", \"{typeof(Emitter).Assembly.GetName().Version}\")]";
+
+    static readonly IndentedStringBuilder IndentedStringBuilder = new();
+
+    static void WriteNamespace(IndentedStringBuilder builder, ClassToGenerate classToGenerate)
+    {
+        if (classToGenerate.Namespace is not null)
+        {
+            builder.Append("namespace ").AppendLine(classToGenerate.Namespace)
```

Comment:
I did a partial revert of the transformation of

```csharp
builder.Append("namespace ").AppendLine(classToGenerate.Namespace)
```

to

```csharp
builder.AppendLine($"namespace {classToGenerate.Namespace}")
```

(and related changes) for performance.

Because source generators must target `netstandard2.0`, the underlying `StringBuilder` doesn't have any interpolated string handling. That means interpolated strings will result in an intermediate allocation.

We _could_ write handling for interpolated strings ourselves, but the code to do so looks nontrivial: https://github.com/dotnet/runtime/pull/51653/files#diff-072e8df6b3d249d58da2b688cbd9369ae5d1e9fb4c1c1da77a19384f3f7711a0.

I also added a comment to the top of `IndentedStringBuilder` explaining the same decision.

Let me know if you strongly disagree and I can try to find another solution.